src/UndockedRegFreeWinRT/UndockedRegFreeWinRT/typeresolution.cpp

```diff
@@ -0,0 +1,657 @@
+                }
+            }
+
+            if ((hr == RO_E_METADATA_NAME_NOT_FOUND) &&
+                (TRO_RESOLVE_NAMESPACE & resolutionOptions))
+            {
+                // Check whether the name is a namespace rather than a type.
+                do
+                {
+                    hr = spMetaDataImport->EnumTypeDefs(
+                        &hEnum,
+                        rgTypeDefs,
+                        ARRAYSIZE(rgTypeDefs),
+                        &cTypeDefs);
+
+                    if (hr == S_OK)
+                    {
+                        for (UINT32 iTokenIndex = 0; iTokenIndex < cTypeDefs; ++iTokenIndex)
+                        {
+                            hr = spMetaDataImport->GetTypeDefProps(
+                                rgTypeDefs[iTokenIndex],
+                                pszRetrievedName,
+                                ARRAYSIZE(pszRetrievedName),
+                                nullptr,
+                                &dwTypeDefProps,
+                                nullptr);
+
+                            if (FAILED(hr))
+                            {
+                                break;
+                            }
+
+                            hr = RO_E_METADATA_NAME_NOT_FOUND;
+
+                            // Only consider windows runtime types when
+                            // trying to determine if the name is a
+                            // namespace.
+                            if (IsTdWindowsRuntime(dwTypeDefProps) && (wcslen(pszRetrievedName) > cFullName))
+                            {
+                                if ((wcsncmp(pszRetrievedName, pszFullName, cFullName) == 0) &&
+                                    (pszRetrievedName[cFullName] == L'.'))
+                                {
+                                    hr = RO_E_METADATA_NAME_IS_NAMESPACE;
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                } while (hr == RO_E_METADATA_NAME_NOT_FOUND);
+
+                // There were no more tokens to enumerate, but the type was still not found.
+                if (hr == S_FALSE)
+                {
+                    hr = RO_E_METADATA_NAME_NOT_FOUND;
+                }
+
+                if (hEnum != nullptr)
+                {
+                    spMetaDataImport->CloseEnum(hEnum);
+                    hEnum = nullptr;
+                }
+            }
+        }
+        return hr;
+    }
+
+    HRESULT FindTypeInDirectory(
+        __in IMetaDataDispenserEx* pMetaDataDispenser,
+        __in PCWSTR pszFullName,
+        __in PCWSTR pszDirectoryPath,
+        __out_opt HSTRING* phstrMetaDataFilePath,
+        __deref_opt_out_opt IMetaDataImport2** ppMetaDataImport,
+        __out_opt mdTypeDef* pmdTypeDef)
+    {
+        HRESULT hr;
+
+        wchar_t szCandidateFilePath[MAX_PATH + 1] = { 0 };
```

Comment:
In GetProcessExeDirInitOnceCallback() you use unique_process_heap_string to account for large filenames.  Do you need to do that here too?

Reply:
I think because were not calling GetModuleFileNameW here that it will be fine?